From 1967c04c021a4cfd7b3cdd4efdc13610b4385a65 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Tue, 11 May 2010 16:36:20 +0200
Subject: [PATCH] Set the close-on-exec flag for listening sockets

This prevents xsm from leaking file descriptors to the processes it
starts.

X.Org bug#22932 <http://bugs.freedesktop.org/show_bug.cgi?id=22932>

Reported-by: Kalle Olavi Niemitalo <kon@iki.fi>
Signed-off-by: Julien Cristau <jcristau@debian.org>
---
 src/listen.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/listen.c b/src/listen.c
index eb46f87..be28aef 100644
--- a/src/listen.c
+++ b/src/listen.c
@@ -76,6 +76,8 @@ IceListenForConnections (
 
     for (i = 0; i < transCount; i++)
     {
+	_IceTransSetOption(transConns[i], TRANS_CLOSEONEXEC, 1);
+
 	networkId = _IceTransGetMyNetworkId (transConns[i]);
 
 	if (networkId)
-- 
1.7.1

